cst438_register/src/main/java/com/cst438/controller/StudentController.java

```diff
@@ -0,0 +1,84 @@
+package com.cst438.controller;
+
+import com.cst438.domain.Student;
+import com.cst438.domain.StudentDTO;
+import com.cst438.domain.StudentRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+public class StudentController {
+
+    @Autowired
+    StudentRepository studentRepository;
+
+    @PostMapping("/student")
+    @Transactional
+    @CrossOrigin(origins = {"http://localhost:3000", "https://registerf-cst438.herokuapp.com/"})
+    
+    public Student addStudent (@RequestBody StudentDTO studentDTO) {
+        
+        Student existStudent = studentRepository.findByEmail(studentDTO.studentEmail);

```

Comment:
This is a smart way of checking if a student exists, then using the result with an if statement to allow for the program to add a student if findByEmail is null